rest_framework_jwt/utils.py

```diff
@@ -23,9 +23,12 @@ def jwt_get_secret_key(payload=None):
     """
     if api_settings.JWT_GET_USER_SECRET_KEY:
         User = get_user_model()  # noqa: N806
-        user = User.objects.get(pk=payload.get('user_id'))
-        key = str(api_settings.JWT_GET_USER_SECRET_KEY(user))
-        return key
+        try:
+            user = User.objects.get(pk=payload.get('user_id'))
+            key = str(api_settings.JWT_GET_USER_SECRET_KEY(user))
```

Comment:
This code should be outside of this `try ... except` block. So it should look like below:

```python
try:
    user = User.objects.get(pk=payload.get('user_id'))
except User.DoesNotExist:
    raise jwt.InvalidTokenError()
else:
    return str(api_settings.JWT_GET_USER_SECRET_KEY(user))
```